js/clinicalTimeline.js

```diff
@@ -163,16 +165,46 @@ var clinicalTimeline = (function(){
       .attr("width", overviewAxisWidth)
       .attr("class", "overview");
 
-    clinicalTimelineOverviewAxis(overviewSVG, getTickValues, minDays, maxDays, overviewAxisWidth, formatTime, daysToTimeObject, getZoomLevel, width,  margin);
-    clinicalTimelineVerticalLine(timelineBool.enableVerticalLine, beginning, ending, timelineBool.advancedView, timelineBool.tooltipOnVerticalLine);
-    initHelperLines(timelineBool.enableHelperLines, zoomFactor, timelineBool.advancedView, divId, formatTime, daysToTimeObject);
+    //object to be shared by all plugins
+    var options = {
+      timelineBool: timelineBool,
+      divId: divId,
+      beginning: beginning,
+      ending: ending,
+      minDays: minDays,
+      maxDays: maxDays,
+      width: width,
+      margin: margin,
+      getZoomLevel: getZoomLevel,
+      getZoomFactor: getZoomFactor,
+      zoomFactor: zoomFactor,
+      getTickValues: getTickValues,
+      formatTime: formatTime,
+      daysToTimeObject: daysToTimeObject,
+      chart: chart,
+      overviewAxisWidth: overviewAxisWidth,
+      overviewX: overviewX,
+      startAllowedOverview: startAllowedOverview,
```

Comment:
`startAllowedOverview` is related to the overview axis plugin, so it should probably reside there. I see it is only used in the `addZoomOptions` function, which probably should be separated out as well. Maybe you can make the addZoomOptions with the overviewAxis a single plugin called `zoom`?